sql/search_aggregates_v5.sql

```diff
@@ -0,0 +1,141 @@
+  udf_mode_last(list ANY TYPE) AS ((
+    SELECT
+      _value
+    FROM
+      UNNEST(list) AS _value
+    WITH
+    OFFSET
+      AS
+    _offset
+    GROUP BY
+      _value
+    ORDER BY
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
+    LIMIT
+      1 ));
+
+WITH
+  summary_addon_version AS (
+  SELECT
+    *,
+    udf_mode_last(ARRAY(
+    SELECT
+      element.version
+    FROM
+      UNNEST(active_addons.list)
+    WHERE
+      element.addon_id = 'followonsearch@mozilla.com')) AS addon_version
+  FROM
+    main_summary_v4
+  ),
+  augmented AS (
+  SELECT
+    s.*,
+    sc.element.source AS source,
+    sc.element.engine AS engine,
+    sc.element.count AS count,
+    CASE
+      WHEN (sc.element.source IN ('searchbar', 'urlbar', 'abouthome', 'newtab', 'contextmenu', 'system', 'activitystream', 'webextension', 'alias') OR sc.element.source IS NULL) THEN 'sap'
+      WHEN STARTS_WITH(sc.element.source, 'in-content:sap:')
+    OR STARTS_WITH(sc.element.source, 'sap:') THEN 'tagged-sap'
+      WHEN STARTS_WITH(sc.element.source, 'in-content:sap-follow-on:') OR STARTS_WITH(sc.element.source,'follow-on:') THEN 'tagged-follow-on'
+      WHEN STARTS_WITH(sc.element.source, 'in-content:organic:') THEN 'organic'
+      WHEN STARTS_WITH(sc.element.source, 'ad-click:') THEN 'ad-click'
+      WHEN STARTS_WITH(sc.element.source, 'search-with-ads:') THEN 'search-with-ads'
+      ELSE 'unknown'
+    END AS type
+
+  FROM
+    summary_addon_version AS s,
+    UNNEST(search_counts.list) AS sc
+  WHERE
+    submission_date_s3 = @submission_date
+  UNION ALL
+  SELECT
+    s.*,
+    "ad-click:" as source,
+    ac.key as engine,
+    ac.value as count,
+    "ad-click" as type
+  FROM
+    summary_addon_version as s,
+    UNNEST(s.scalar_parent_browser_search_ad_clicks.key_value) as ac
+  WHERE
+    s.scalar_parent_browser_search_ad_clicks IS NOT NULL
+  UNION ALL
+  SELECT
+    s.*,
+    "search-with-ads:" as source,
+    ac.key as engine,
+    ac.value as count,
+    "search-with-ads" as type
+  FROM
+    summary_addon_version as s,
+    UNNEST(s.scalar_parent_browser_search_with_ads.key_value) as ac
+  WHERE
+    s.scalar_parent_browser_search_with_ads IS NOT NULL
+  ),
+  aggregated AS (
+  SELECT
+    submission_date_s3 AS submission_date,
+    addon_version,
+    app_version,
+    country,
+    distribution_id,
+    locale,
+    search_cohort,
+    engine,
+    source,
+    SUM(count) AS count,
+    default_search_engine,
+    type
+  FROM
+    augmented
+  WHERE
+    count < 10000
+    AND engine IS NOT NULL
+  GROUP BY
+    submission_date_s3,
+    addon_version,
+    app_version,
+    country,
+    distribution_id,
+    engine,
+    locale,
+    search_cohort,
+    source,
+    default_search_engine,
+    type)
+SELECT
+  submission_date,
+  addon_version,
+  app_version,
+  country,
+  distribution_id,
+  engine,
+  locale,
+  search_cohort,
+  source,
+  default_search_engine,
+  NULLIF(SUM(IF(type = 'organic', count, 0)), 0) AS organic,
```

Comment:
The use of `NULLIF` here seems odd, is there a reason we want this to have `NULL`s instead of `0`s? if not we can actually differentiate between `NULL` and `0` by doing this: 
```suggestion
  SUM(IF(type = 'organic', count, NULL)) AS organic,
```

Reply:
cc @harterrt because git blame tells me this was your comment: https://github.com/mozilla/python_mozetl/blob/master/mozetl/search/aggregates.py#L278-L279

should we instead be making sure we *don't* have nulls via:
```suggestion
  SUM(IF(type = 'organic', count, 0)) AS organic,
```

Reply:
`SUM(IF(type = 'organic', count, NULL)) AS organic,` seems right to me. The PR you linked to primarily affects `search_clients_daily` and adds placeholders for users who had pings without `search_counts` (i.e. zero-search-users).

Will you check for parity for a few days data after merging this? These datasets have a lot of context baked in.

Reply:
cc @wlach for context